pkg/apis/volumesnapshot/v1beta1/types.go

```diff
@@ -0,0 +1,354 @@
+type VolumeSnapshotContentList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// items is the list of VolumeSnapshotContents
+	Items []VolumeSnapshotContent `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// VolumeSnapshotContentSpec is the specification of a VolumeSnapshotContent
+type VolumeSnapshotContentSpec struct {
+	// volumeSnapshotRef specifies the VolumeSnapshot object to which this
+	// VolumeSnapshotContent object is bound.
+	// VolumeSnapshot.Spec.VolumeSnapshotContentName field must reference to
+	// this VolumeSnapshotContent's name for the bidirectional binding to be valid.
+	// For a pre-existing VolumeSnapshotContent object, name and namespace of the
+	// VolumeSnapshot object MUST be provided for binding to happen.
+	// Required.
+	VolumeSnapshotRef core_v1.ObjectReference `json:"volumeSnapshotRef" protobuf:"bytes,1,opt,name=volumeSnapshotRef"`
+
+	// deletionPolicy determines whether this VolumeSnapshotContent and its physical snapshot on
+	// the underlying storage system should be deleted when its bound VolumeSnapshot is deleted.
+	// Supported values are "Retain" and "Delete".
+	// "Retain" means that the VolumeSnapshotContent and its physical snapshot on underlying storage system are kept.
+	// "Delete" means that the VolumeSnapshotContent and its physical snapshot on underlying storage system are deleted.
+	// In dynamic snapshot creation case, this field will be filled in with the "DeletionPolicy" field defined in the
+	// VolumeSnapshotClass the VolumeSnapshot refers to.
+	// For pre-existing snapshots, users MUST specify this field when creating the VolumeSnapshotContent object.
+	// +kubebuilder:validation:Enum=Delete;Retain
+	// Required.
+	DeletionPolicy DeletionPolicy `json:"deletionPolicy" protobuf:"bytes,2,opt,name=deletionPolicy"`
+
+	// driver is the name of the CSI driver used to create the physical snapshot on
+	// the underlying storage system.
+	// This MUST be the same as the name returned by the CSI GetPluginName() call for
+	// that driver.
+	// Required.
+	Driver string `json:"driver" protobuf:"bytes,3,opt,name=driver"`
+
+	// name of the SnapshotClass to which this snapshot belongs.
+	// +optional
+	SnapshotClassName *string `json:"snapshotClassName,omitempty" protobuf:"bytes,4,opt,name=snapshotClassName"`
+
+	// source specifies from where a snapshot will be created.
+	// Required.
+	Source VolumeSnapshotContentSource `json:"source" protobuf:"bytes,5,opt,name=source"`
+}
+
+// VolumeSnapshotContentSource represents the source of a snapshot.
+// Exactly one of its members must be set.
+// Members in VolumeSnapshotContentSource are immutable.
+// TODO(xiangqian): Add a webhook to ensure that VolumeSnapshotContentSource members
+// will not be updated once specified.
+type VolumeSnapshotContentSource struct {
+	// volumeHandle specifies the CSI name of the volume from which a snapshot
```

Comment:
use the csi spec names here? ie "volume_id" and "snapshot_id"

Reply:
VolumeHandle/SnapshotHandle seems to be the convention in other places. I think it's better to keep it consistent?

Reply:
In the comments, refer to the actual csi spec names instead of "CSI name of the volume/snapshot", etc.

Reply:
updated

Reply:
don't see an update?

Reply:
hmm, missed this, sry. Updated.